da/grpc/grpc.go

```diff
@@ -0,0 +1,103 @@
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"google.golang.org/grpc"
+
+	"github.com/celestiaorg/optimint/da"
+	"github.com/celestiaorg/optimint/log"
+	"github.com/celestiaorg/optimint/store"
+	"github.com/celestiaorg/optimint/types"
+	"github.com/celestiaorg/optimint/types/pb/dalc"
+)
+
+type DataAvailabilityLayerClient struct {
+	config Config
+
+	conn   *grpc.ClientConn
+	client dalc.DataAvailabilityLayerClientServiceClient
```

Comment:
Orthogonal to this PR but here the name is too Java-style longish especially for an interface! This is not very golang idiomatic!